src/H5Dselect.c

```diff
@@ -256,6 +256,187 @@ H5D__select_io(const H5D_io_info_t *io_info, size_t elmt_size, size_t nelmts, co
     FUNC_LEAVE_NOAPI(ret_value)
 } /* end H5D__select_io() */
 
+/*-------------------------------------------------------------------------
+ * Function:    H5D_select_io_mem
+ *
+ * Purpose:     Perform memory copies directly between two memory buffers
+ *              according to the selections in the `dst_space` and
+ *              `src_space` dataspaces.
+ *
+ * Note:        This routine is [basically] the same as H5D__select_io,
+ *              with the only difference being that the readvv/writevv
+ *              calls are exchanged for H5VM_memcpyvv calls. Changes should
+ *              be made to both routines.
+ *
+ * Return:      Non-negative on success/Negative on failure
+ *
+ *-------------------------------------------------------------------------
+ */
+herr_t
+H5D_select_io_mem(void *dst_buf, const H5S_t *dst_space, const void *src_buf, const H5S_t *src_space,
```

Comment:
A new routine that is very similar to H5D__select_io(), but rather than copying between application memory and the file, copies between two memory buffers according to the selection in the dst and src dataspaces.